kernel/classes/datatypes/ezimage/ezimagealiashandler.php

```diff
@@ -378,13 +378,36 @@ function imagePathByNode( $contentObjectAttribute, $mainNode )
         $pathParts = array( eZSys::storageDirectory(), $contentImageSubtree );
         if ( $pathString != '' )
         {
-            $pathParts[] = $pathString;
+            //Make sure that all folders are smaller than the FS limit
+            foreach ( explode( '/', $pathString ) as $folder )
+            {
+                $pathParts[] = self::trimToFileSystemFileName( $folder );
+            }
         }
         $pathParts[] = $attributeID . '-' . $attributeVersion . '-' . $attributeLanguage;
         $imagePath = implode( '/', $pathParts );
         return $imagePath;
     }
 
+    /**
+     * Trims a filename to a given limit (in bytes).
+     * It can be used to avoid file system limitations
+     *
+     * @param $longName
+     * @param int $limit in bytes
+     *
+     * @return string shortened name
+     */
+    public static function trimToFileSystemFileName( $longName, $limit = 200 )
```

Comment:
Unless I've missed something, you don't have to make it static (but it should be public). `imageNameByNode` already uses `$this`, and `imagePathByNode` isn't called statically:

```
$ git grep imagePathByNode
kernel/classes/datatypes/ezimage/ezimagealiashandler.php:    function imagePathByNode( $contentObjectAttribute, $mainNode )
kernel/classes/datatypes/ezimage/ezimagetype.php:                $dirpath = $imageHandler->imagePathByNode( $contentObjectAttribute, $mainNode );
```